src/main/java/net/dv8tion/jda/api/interactions/components/ButtonInteraction.java

```diff
@@ -79,7 +79,7 @@ default Button getComponent()
             for (ListIterator<Component> it = row.listIterator(); it.hasNext();)
             {
                 Component component = it.next();
-                if (id.equals(component.getId()))
+                if (component.getId().equals(newButton.getId()))
```

Comment:
This is actually wrong. This method intentionally does NOT check the id of newButton because it is valid to change the button id.